provisioning/service/samples/bulkupdate_sample.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var assert = require('chai').assert;
+
+var provisioningClient = require('azure-iot-provisioning-service-client');
+
+var serviceClient = provisioningClient.DeviceEnrollment.fromConnectionString(process.argv[2]);
+
+var enrollment1 = {
+  registrationId: 'first',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+var enrollment2 = {
+  registrationId: 'second',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+var enrollment3 = {
+  registrationId: 'third',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm'
+  }
+};
+
+var badCreateBulkOperation = {
+  mode: 'create',
+  enrollments: []
+};
+badCreateBulkOperation.enrollments.push(enrollment1);
+badCreateBulkOperation.enrollments.push(enrollment2);
+badCreateBulkOperation.enrollments.push(enrollment2);
+
+//
+// This bulk operation will fail.  The reason it fails is because we have
+// a duplicate entry in the bulk array.  This will NOT produce a BulkOperationResult.
+//
+serviceClient.bulkOperation(badCreateBulkOperation, function(err, result, response) {
+  if (err) {
+    console.log('The bulk operation result is: ' + err);
+  }
+  console.log('The value of the result operation should be undefined: ',JSON.stringify(result, null, 2));
+  //
+  // Get rid of the duplicate entry and put in an entry without an actual attestation.
+  // Some of the bulk create will work but we should get a bulk operation result
+  // that indicates the third one failed.
+  //
+  badCreateBulkOperation.enrollments.splice(2,1);
+  badCreateBulkOperation.enrollments.push(enrollment3);
+  serviceClient.bulkOperation(badCreateBulkOperation, function(err, result, response) {
+    console.log('The err parameter should be null for this second bulkOperation call, err: ' + err);
+    console.log('The value of the result operation should indicate the third enrollment was bad: ',JSON.stringify(result, null, 2));
+    //
+    // Bulk operations don't give you back the "filled out" enrollment object.  To get
+    // that we either need to do a query or a straight get.  Let's use get.
+    //
```

Comment:
why? feels like a passive aggressive way of critiquing the service design. samples are not meant for that. if you want to point out an API design choice we made, let's explain why we're making it that way

Reply:
Removed comment.

---
In reply to: [147066827](https://github.com/Azure/azure-iot-sdk-node/pull/121#discussion_r147066827) [](ancestors = 147066827)